frame/staking/src/lib.rs

```diff
@@ -668,8 +675,17 @@ decl_storage! {
 		/// Nominators for a particular account that is in action right now. You can't iterate
 		/// through validators here, but you can find them in the Session module.
 		///
-		/// This is keyed by the stash account.
-		pub Stakers get(fn stakers): map T::AccountId => Exposure<T::AccountId, BalanceOf<T>>;
+		/// This is keyed fist by the era index to allow bulk deletion and then the stash account.
+		///
+		/// Is it removed after `HISTORY_DEPTH` eras.
+		pub Stakers get(fn stakers): double_map EraIndex hasher(twox_64_concat),
+			twox_64_concat(T::AccountId) => Exposure<T::AccountId, BalanceOf<T>>;
```

Comment:
I was wondering if having a `Vec<(T::Account, Exposure<..>)>` for historical stakers exposure was not better here, but iterating on prefix could be already efficient enough.